rfcs/20240903-onemkl-interface-rename/README.md

```diff
@@ -0,0 +1,205 @@
+oneMKL Interface. The oneMKL Interface
+[README](https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#onemkl) already
+explains the differences between oneMKL Interface and Intel oneMKL product but
+this is not enough for people who are not working directly on the projects.
+
+## Outline
+
+1. [Introduction](#introduction)
+2. [Proposal](#proposal)
+3. [User impact](#user-impact)
+4. [Open questions](#open-questions)
+
+## Introduction
+
+oneMKL Interface has historically been implemented and supported by Intel. As
+the project is moving to the UXL foundation we want to avoid using the name of
+an Intel product. This RFC describes the solution of renaming oneMKL Interface
+to a new name.
+
+## Proposal
+
+The main purpose of this RFC is to agree on a new name for the oneMKL specification and oneMKL Interface.
+Some of the suggested names for the implementations are:
+* **oneMath**
+  * The specification would be renamed oneMath Specification
+  * The C++ namespace would be `oneapi::math`
+  * The CMake namespace would be `ONEMATH`
+* **oneSLA** (SYCL Linear Algebra)
+  * The specification would be renamed oneSLA Specification
+  * New namespace would be `oneapi::sla`
+  * The CMake namespace would be `ONESLA`
+
+Other suggestions are welcomed. The name **oneMath** will be chosen if there are
+no objections by October 4, 2024.
+
+The suggested solution is to proceed in the following steps:
+1. The UXL foundation agrees on the new name.
+2. Codeplay submits a oneAPI-spec PR to rename the occurrences of "oneMKL" and
+   "mkl" to the new name.
+3. Codeplay submits a oneMKL Interface PR to:
+  * Update the root README to use the new name, with a mention that the project
+    was formerly called oneMKL Interface.
+  * Update the references to "oneMKL" and `onemkl_` in the documentation as
+    seen in the first few lines of
+    [docs/onemkl-datatypes.rst](https://github.com/oneapi-src/oneMKL/blob/develop/docs/onemkl-datatypes.rst?plain=1#L1)
+    for instance.
+  * Update occurrences of "onemkl" in internal functions such as
+    [onemkl_cublas_host_task](https://github.com/oneapi-src/oneMKL/blob/1ce98a699f93bd3a78350269b2e34d822fe43b91/src/blas/backends/cublas/cublas_task.hpp#L77).
+  * Update macros such as include guards and other internal macros like
+    `ONEMKL_EXPORT` to use the new name.
+  * Rename CMake targets `onemkl` and `onemkl_<domain>_<backend>` to use the
+    new name. The existing targets name can be added with a deprecation
+    messages for anyone using them. See the section on [CMake target
+    deprecation](#cmake-deprecated-target) for more details. The namespace of
+    the exported and installed targets are changed to use the new name.
+  * The namespace `oneapi::mkl` is deprecated and aliased to the new namespace
+     name.
+  * The main header and domain headers are deprecated and include the new headers as shown below:
+    * `include/oneapi/`
+      * `mkl.hpp` -> `onemath.hpp`
+      * `mkl/` -> `onemath/`
```

Comment:
why onemath.hpp and not math.hpp ? and math/ instead of onemath/ ?  since it is in the include/oneapi/ folder where the "one" presumably comes from ?

Reply:
I have a small preference for `onemath` as I find `math` is just too generic. I don't think it is an issue to repeat the `one` prefix. I have renamed it to https://github.com/oneapi-src/oneMKL/pull/564/commits/1182a951fb51a347a2a0a40437d430a759bdf55e since I seem outvoted for now.

Reply:
Other oneAPI parts don't repeat the prefix: [`<oneapi/dpl/execution>`](https://oneapi-spec.uxlfoundation.org/specifications/oneapi/v1.4-provisional-rev-1/elements/onedpl/source/parallel_api/execution_policies), [`"oneapi/dal.hpp"`](https://oneapi-spec.uxlfoundation.org/specifications/oneapi/v1.4-provisional-rev-1/elements/onedal/source/programming_model/end_to_end_example), [`<oneapi/tbb/collaborative_call_once.h>`](https://oneapi-spec.uxlfoundation.org/specifications/oneapi/v1.4-provisional-rev-1/elements/onetbb/source/algorithms/functions/collaborative_call_once_func).